src/LegacyStrings.jl

```diff
@@ -49,15 +49,15 @@ using Compat
     end
 
     if VERSION >= v"0.5.0-"
-        immutable ASCIIString <: DirectIndexString
+        struct ASCIIString <: DirectIndexString
```

Comment:
This package still supports 0.4 but `struct` only parses properly on 0.6+.

Reply:
StringEncodings uses LegacyStrings mainly as a way exercize support for string types in different encodings. I could easily remove it if LegacyStrings is declared obsolete.

OTOH, if that's not too much work keeping the package alive makes sense, as these string types can be useful for specialized use cases. Of course somebody could revive it if it's really needed.